fmt/container.h

```diff
@@ -0,0 +1,81 @@
+/*
+ Formatting library for C++ - standard container utilities
+
+ Copyright (c) 2012 - 2016, Victor Zverovich
+ All rights reserved.
+
+ For the license information refer to format.h.
+ */
+
+#ifndef FMT_CONTAINER_H_
+#define FMT_CONTAINER_H_
+
+#include "format.h"
+
+namespace fmt {
+
+namespace internal {
+
+/**
+  \rst
+  A "buffer" that appends data to a standard container (e.g. typically a
+  ``std::vector`` or ``std::basic_string``).
+  \endrst
+ */
+template <typename Container>
+class ContainerBuffer : public Buffer<typename Container::value_type> {
+ private:
+  Container& container_;
+  std::size_t initial_size_;
+
+ protected:
+  virtual void grow(std::size_t size) FMT_OVERRIDE {
+    container_.resize(initial_size_ + size);
+    this->ptr_ = &container_[initial_size_];
+    this->capacity_ = size;
+  }
+
+ public:
+  explicit ContainerBuffer(Container& container)
+  : container_(container), initial_size_(container.size()) {}
```

Comment:
Then I'm not sure sure how `grow()` should work in case `container_` is not empty (i.e. in append mode)? Even if I initialize `size_` and `capacity_` from the constructor, I fail to see how I can initialize `ptr_` safely from there. Did I miss something?